### ISSUE TITLE
feature request: not require actor versions below network's genesis version

### ISSUE THREAD

**Issue by raulk** [filecoin-project/lotus#9677]
We’ve hit some Lotus limitations today during our Topaz release to Wallaby.

- The Wallaby testnet’s genesis starts at nv18 (actors v10).
- Unfortunately, Lotus today expects bundles to be available for all networks for all actor versions.
- This means that we’d had to force the generation of actors v8 and actors v9 bundles, even if the network will never use them.
- [NB: there may be some use for v9 bundles if we test the migration in Wallaby.]

The question is: should we adjust Lotus so that it doesn’t expect actor bundles below the genesis version of the network?

**Comment by vyzo** [comment]
I think we should, it is wuite broken as it is right now.

**Comment by arajasek** [comment]
Yeah, I'm in favour of doing this. It should be relatively simple to do, but there might be obstacles hiding.

I think this is part of a broader stream of work needed to make Lotus more flexible across networks. In addition to not requiring earlier actors versions, we should also make it possible to have different upgrade schedules & migrations, land #8158, break connections between LegacyVM and FVM code so that networks starting from v16 and later can be cleanly FVM-only.
